src/ArgsAwareTrait.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Dhii\Invocation;
+
+use Exception as RootException;
+use InvalidArgumentException;
+
+/**
+ * Functionality for args awareness.
+ *
+ * @since [*next-version*]
+ */
+trait ArgsAwareTrait
+{
+    /**
+     * The args.
+     *
+     * @since [*next-version*]
+     *
+     * @var array
+     */
+    protected $args;
+
+    /**
+     * Retrieves the args associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return array The args.
+     */
+    protected function _getArgs()
+    {
+        return $this->args;
+    }
+
+    /**
+     * Assigns a args to this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @param array $args The args to assign.
```

Comment:
I feel like this, and maybe also the property of this trait, should document what the array is expected to contain, rather than what the parameter signifies. For instance, "an array of argument values" or "a mapping of argument names to values".